diff --git a/ggml/src/ggml-backend.cpp b/ggml/src/ggml-backend.cpp
index 8761592..639f571 100644
--- a/ggml/src/ggml-backend.cpp
+++ b/ggml/src/ggml-backend.cpp
@@ -750,7 +750,7 @@ static bool ggml_is_view_op(enum ggml_op op) {
 // scheduler
 
 #ifndef GGML_SCHED_MAX_BACKENDS
-#define GGML_SCHED_MAX_BACKENDS 16
+#define GGML_SCHED_MAX_BACKENDS 32
 #endif
 
 #ifndef GGML_SCHED_MAX_SPLIT_INPUTS
diff --git a/src/llama.cpp b/src/llama.cpp
index 0de6048..468baa4 100644
--- a/src/llama.cpp
+++ b/src/llama.cpp
@@ -55,7 +55,7 @@ struct llama_sampler_chain_params llama_sampler_chain_default_params() {
 }
 
 size_t llama_max_devices(void) {
-    return 16;
+    return 32;
 }
 
 size_t llama_max_tensor_buft_overrides() {
